test/controllers/selectDirector.controller.test.ts

```diff
@@ -312,7 +364,7 @@ describe('SelectDirectorController', () => {
           { ...generateDirectorDetails(), id: DIRECTOR_1_ID },
           { ...generateDirectorDetails(), id: DIRECTOR_2_ID }
         ])
-        when(validator.validate(deepEqual(form), selectDirectorSchema)).thenReturn(null)
+        when(validator.validate(deepEqual(form), selectDirectorSchema(dissolutionSession.officerType!))).thenReturn(null)
```

Comment:
This when condition is not being satisfied. `selectDirectorSchema(dissolutionSession.officerType!)` is created a new schema object, which will be different from the one in the implementation. Wrap this in a `deepEqual` instead.